ir/src/main/scala/org/scalajs/core/ir/Trees.scala

```diff
@@ -965,9 +965,11 @@ object Trees {
 
   }
 
-  /** A hash of a tree (usually a MethodDef). Contains two SHA-1 hashes */
-  final class TreeHash(val treeHash: Array[Byte], val posHash: Array[Byte]) {
-    assert(treeHash.length == 20)
-    assert(posHash.length == 20)
+  /** A hash of a tree (usually a MethodDef).
+   *
+   *  Contains a SHA-1 hash.
+   */
+  final class TreeHash(val hash: Array[Byte]) {
```

Comment:
It can probably be `AnyVal`, indeed. I'll have to check whether it would be boxed more often than not, but I don't think so.

Inlining would make it more annoying to ensure invariants, not to mention being typesafe for methods like `hashesEqual` and `hashAsVersion`.